client/src/legacy/GridField.js

```diff
@@ -180,6 +194,19 @@ $.entwine('ss', function($) {
       if (hasLegacyFilterHeader) {
         this.find('.sortable-header th:last').html(content);
       }
+    },
+
+    keepStateInHistory: function() {
+      const newURLSceme = $(this).find('.gridfield-actionmenu__container').data('schema');
+      if (newURLSceme && newURLSceme.length > 0) {
+        newURLSceme.filter( e => {
+          if (e.type === 'link') {
+            const searchParam = e.url.includes('?') ? '?' + e.url.split('?')[1] : '';
+            const historyState = $.extend({}, {path: window.location.pathname + searchParam}, this.getState());
+            history.replaceState(historyState, '', window.location.pathname + searchParam);
```

Comment:
Make sure we're not just throwing away other GET vars that were already present

Reply:
Thank you. I'll check parameters that we pass in `GridField.php` class. Since, here I just grab all search params from scheme, that we also pass to Edit button and as URL for click. So, would be better to be sure that we have a correct value in the scheme.
What do you think?

Reply:
This is the URL for the gridfield we're currently looking at - which needs to include any GET parameters that are _already_ in the URL.
So for example if I'm on some page `/admin/pages/edit/show/2?some-var=some-value` and I click paginate on a gridfield, the `some-var=some-value` shouldn't be removed. These may be here for any reason beyond gridfield, and removing them may cause problems in some project-specific code that relies on them for whatever reason.

The edit and view buttons don't need these, because they're relevant only while you're on the page that holds the gridfield, _not_ for the item edit form.